gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_sdk_custom_metric.ipynb

```diff
@@ -0,0 +1,1390 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1wGoVQNCMbxe"
+      },
+      "outputs": [],
+      "source": [
+        "custom_metric_function_code = '''\n",
+        "\"\"\"\n",
+        "This module contains the custom evaluation metric definition to optimize a prompt template with Vertex AI Prompt Optimizer\n",
+        "\"\"\"\n",
+        "\n",
+        "from typing import Dict\n",
+        "from vertexai.generative_models import (\n",
+        "    GenerationConfig,\n",
+        "    GenerativeModel,\n",
+        "    HarmBlockThreshold,\n",
+        "    HarmCategory,\n",
+        ")\n",
+        "\n",
+        "import json\n",
+        "import functions_framework\n",
+        "\n",
+        "def get_autorater_response(metric_prompt: str) -> dict:\n",
+        "    \"\"\"This function is to generate the evaluation response from the autorater.\"\"\"\n",
+        "\n",
+        "    metric_response_schema = {\n",
+        "        \"type\": \"OBJECT\",\n",
+        "        \"properties\": {\n",
+        "            \"score\": {\"type\": \"NUMBER\"},\n",
+        "            \"explanation\": {\"type\": \"STRING\"},\n",
+        "        },\n",
+        "        \"required\": [\"score\", \"explanation\"],\n",
+        "    }\n",
+        "\n",
+        "    autorater = GenerativeModel(\n",
+        "        \"gemini-1.5-pro\",\n",
+        "        generation_config=GenerationConfig(\n",
+        "            response_mime_type=\"application/json\",\n",
+        "            response_schema=metric_response_schema,\n",
+        "        ),\n",
+        "        safety_settings={\n",
+        "            HarmCategory.HARM_CATEGORY_UNSPECIFIED: HarmBlockThreshold.BLOCK_NONE,\n",
+        "            HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_NONE,\n",
+        "            HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_NONE,\n",
+        "            HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_NONE,\n",
+        "            HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: HarmBlockThreshold.BLOCK_NONE,\n",
+        "        },\n",
+        "    )\n",
+        "\n",
+        "    response = autorater.generate_content(metric_prompt)\n",
+        "\n",
+        "    response_json = {}\n",
+        "\n",
+        "    if response.candidates and len(response.candidates) > 0:\n",
+        "        candidate = response.candidates[0]\n",
+        "        if (\n",
+        "            candidate.content\n",
+        "            and candidate.content.parts\n",
+        "            and len(candidate.content.parts) > 0\n",
+        "        ):\n",
+        "            part = candidate.content.parts[0]\n",
+        "            if part.text:\n",
+        "                response_json = json.loads(part.text)\n",
+        "\n",
```

Comment:
I think you can just access `response.text` and the Vertex AI SDK does all of these checks.

```py
return json.loads(response.text)
```

Reply:
done